src/Analyser/Comment/CommentParser.php

```diff
@@ -47,10 +49,26 @@ public function parseIgnoreComment(Comment $comment, Node $node): ?IgnoreComment
 		);
 
 		if (count($ignoreMessageRegexpMatches) > 0) {
+			$this->validateNode($comment, $node);
 			return IgnoreComment::createIgnoreRegexp($comment, $node, trim($ignoreMessageRegexpMatches[1]));
 		}
 
 		return null;
 	}
 
+	private function validateNode(Comment $comment, Node $node): void
+	{
+		$invalidNodes = [
+			Node\Stmt\ClassLike::class,
+			Node\Stmt\ClassMethod::class,
+			Node\Stmt\Function_::class,
+		];
+
+		foreach ($invalidNodes as $invalidNode) {
+			if ($node instanceof $invalidNode) {
+				throw new \PHPStan\Analyser\Comment\Exception\InvalidIgnoreNextLineNodeException($comment, $node->getType());
```

Comment:
Is this commit necessary? I don't think so. As long as `next-line` works properly (for the next line only), this usage can be valid...

Reply:
For example it could be used to ignore some message about a function parameter...

Reply:
I wasn't sure if there can be PHPStan errors in function parameters or classes. Do you have an example for me?
If there's the possibility then I'll of course remove this check

Reply:
Yeah, of course: https://phpstan.org/r/324360c1-4561-44f6-a99a-dbe296141065